tests/validation/tests/v3_api/test_global_role.py

```diff
@@ -0,0 +1,348 @@
+    token = rbac_get_user_token_by_role(CLUSTER_OWNER)
+    validate_edit_global_role(token, gr, False)
+
+
+@if_test_rbac
+def test_rbac_global_role_edit_cluster_member(remove_resource):
+    gr = create_gr()
+    remove_resource(gr)
+    token = rbac_get_user_token_by_role(CLUSTER_MEMBER)
+    validate_edit_global_role(token, gr, False)
+
+
+@if_test_rbac
+def test_rbac_global_role_edit_project_owner(remove_resource):
+    gr = create_gr()
+    remove_resource(gr)
+    token = rbac_get_user_token_by_role(PROJECT_OWNER)
+    validate_edit_global_role(token, gr, False)
+
+
+@if_test_rbac
+def test_rbac_global_role_edit_project_member(remove_resource):
+    gr = create_gr()
+    remove_resource(gr)
+    token = rbac_get_user_token_by_role(PROJECT_MEMBER)
+    validate_edit_global_role(token, gr, False)
+
+
+@if_test_rbac
+def test_rbac_global_role_edit_project_read_only(remove_resource):
+    gr = create_gr()
+    remove_resource(gr)
+    token = rbac_get_user_token_by_role(PROJECT_READ_ONLY)
+    validate_edit_global_role(token, gr, False)
+
+
+@pytest.fixture(scope='module', autouse="True")
+def create_project_client(request):
+    client, cluster = get_global_admin_client_and_cluster()
+    create_kubeconfig(cluster)
+
+
+def create_gr():
+    return validate_create_global_role(ADMIN_TOKEN, False, True)
+
+
+def generate_template(name, new_user_default=False):
+    template = deepcopy(template_manage_catalog)
+    if new_user_default:
+        template["newUserDefault"] = "true"
+    else:
+        template["newUserDefault"] = "false"
+    if name is None:
+        name = random_name()
+    template["name"] = name
+    return template
+
+
+def validate_create_global_role(token, new_user_default=False,
+                                permission=False):
+    client = get_client_for_token(token)
+    t_name = random_name()
+    template = generate_template(t_name, new_user_default)
+    try:
+        client.create_global_role(template)
+    except ApiError as e:
+        if not permission:
+            assert e.error.status == 403
+            assert e.error.code == 'Forbidden'
+        else:
+            assert False, "non 403 error is returned"
+        return None
+
+    # check that the global role is created
+    gr_list = client.list_global_role(name=t_name).data
+    assert len(gr_list) == 1
+    gr = gr_list[0]
+    # check that the global role is set to be the default
+    assert gr.newUserDefault == new_user_default
+    return gr
+
+
+def validate_delete_global_role(token, global_role, permission=False):
+    client = get_client_for_token(token)
+    try:
+        client.delete(global_role)
+    except ApiError as e:
+        if not permission:
+            assert e.error.status == 403
+            assert e.error.code == 'Forbidden'
+        else:
+            assert False, "non 403 error is returned"
+        return None
+    client = get_client_for_token(ADMIN_TOKEN)
+    assert client.reload(global_role) is None
```

Comment:
Why a reload is required instead of a "list" ?

Reply:
they do the same thing:
```
reload(foo) == list(type=foo.tyep, id=foo.id)
```